src/components/store/reducer.ts

```diff
@@ -0,0 +1,30 @@
+import { EditorState } from '../../../types/store/editorState';
+import { Action } from '../../../types/store/action';
+import * as _ from '../utils';
+import { BlockMutationType } from '../../../types/events/block/mutation-type';
+
+/**
+ * Reducer function for Editor.js state
+ * This function applies the passed action to the current state and returns the new state
+ *
+ * @param state - previous state to apply action
+ * @param action - information about the action in the previous state
+ */
+const reducer = (state: EditorState, action: Action): EditorState => {
```

Comment:
also, naming is not clear. If you're talking that we can have several different reducers, so we need to call this reducer somehow more specific.

Along with that, looks like this reducer works not with the whole Editor State, but only with the `blocks` property. 

Maybe it would better to call it `BlocksReducer`?

Reply:
Yep, it's a good idea, I will change it